tedana/io.py

```diff
@@ -301,65 +488,31 @@ def writeresults_echoes(catd, mmix, mask, comptable, ref_img, out_dir='.'):
     for i_echo in range(catd.shape[1]):
         LGR.info('Writing Kappa-filtered echo #{:01d} timeseries'.format(i_echo + 1))
         write_split_ts(
-            catd[:, i_echo, :], mmix, mask, comptable, ref_img,
-            out_dir=out_dir,
-            prefix='echo-{}_desc-'.format(i_echo + 1)
+                catd[:, i_echo, :], mmix, mask, comptable, ref_img,
+                echo=(i_echo + 1)
         )
 
 
-def new_nii_like(ref_img, data, affine=None, copy_header=True):
-    """
-    Coerces `data` into NiftiImage format like `ref_img`
-
-    Parameters
-    ----------
-    ref_img : :obj:`str` or img_like
-        Reference image
-    data : (S [x T]) array_like
-        Data to be saved
-    affine : (4 x 4) array_like, optional
-        Transformation matrix to be used. Default: `ref_img.affine`
-    copy_header : :obj:`bool`, optional
-        Whether to copy header from `ref_img` to new image. Default: True
-
-    Returns
-    -------
-    nii : :obj:`nibabel.nifti1.Nifti1Image`
-        NiftiImage
-    """
-
-    ref_img = check_niimg(ref_img)
-    newdata = data.reshape(ref_img.shape[:3] + data.shape[1:])
-    if '.nii' not in ref_img.valid_exts:
-        # this is rather ugly and may lose some information...
-        nii = nib.Nifti1Image(newdata, affine=ref_img.affine,
-                              header=ref_img.header)
-    else:
-        # nilearn's `new_img_like` is a very nice function
-        nii = new_img_like(ref_img, newdata, affine=affine,
-                           copy_header=copy_header)
-    nii.set_data_dtype(data.dtype)
-
-    return nii
-
-
-def filewrite(data, filename, ref_img, gzip=True, copy_header=True):
+def filewrite(data, img_type, ref_img, gzip=True, copy_header=True,
+              echo=0):
     """
     Writes `data` to `filename` in format of `ref_img`
 
     Parameters
     ----------
     data : (S [x T]) array_like
         Data to be saved
-    filename : :obj:`str`
-        Filepath where data should be saved to
+    img_type : :obj: `str`
```

Comment:
Adding a space here breaks the rendering. Compare `img_type` and `ref_img` in the RTD CI build:

![image](https://user-images.githubusercontent.com/8228902/112190987-eae66b00-8bdb-11eb-97cc-02aa274353f4.png)